### ISSUE TITLE
Enhance, Cleanup and Clarify Documentation for using REX-Ray with Azure Storage

### ISSUE THREAD

**Issue by jzampieron** [codedellemc/libstorage#452]
# Summary
It is exceedingly difficult to setup _both_ REX-Ray and your azure account to use the Azure UD driver with the currently published documentation.

The error messages from REX-Ray are not obvious and the documentation does not reflect the same terminology. 

# Bug Reports
Following this: http://libstorage.readthedocs.io/en/stable/user-guide/storage-providers/#microsoft and trying to use the config.yml file specified results in REX-Ray failing to be able to successfully access azure. (Rexray 0.8.1 installed from deb files on Ubuntu 16.04 - Azure VM.)

## Expected Behavior
Step by Step directions with screen shots for configuring an "App Registration" via the Azure Portal for use with Rexray. A config.yml for Rexray 0.8.1 or newer that is usable. 

Documentation terminology for the `azureud` driver configuration should reflect the MSFT Azure Portal terminology, i.e. `clientId` is called `Application Id` in the App Registration, etc.

## Actual Behavior
`rexray volume ls` command fails.

## Steps To Reproduce
Please list the steps to reproduce the issue in this section.

Apply the recommended configuration file:
```yaml
libstorage:
  server:
    tasks:
      exeTimeout: 120s
    services:
      azure:``
        driver: azureud
        azureud:
          subscriptionID: abcdef01-2345-6789-abcd-ef0123456789
          resourceGroup: testgroup
          tenantID: usernamehotmail.onmicrosoft.com
          storageAccount: username
          storageAccessKey: XXXXXXXX
          clientID: 123def01-2345-6789-abcd-ef0123456789
          clientSecret: XXXXXXXX
```

Note that Volume LS and other commands fail.

Applying the following configuration file partially works (create and ls):

```yaml
rexray:
  logLevel: info
  modules:
    #default-admin:
    #  host: tcp://127.0.0.1:61003
    default-docker:
      disabled: true
libstorage:
  server:
    tasks:
      exeTimeout: 120s
  service: azureud
azureud:
  subscriptionID: subuuidher
  resourceGroup: resourcegrouphere
  tenantID: tenantidhere
  storageAccount: storageaccounthere
  storageAccessKey: KEYHERE
  clientID: appregistrationApplicationIdHere
  clientSecret: appRegistrationSecretHere
  useHTTPS: true
  container: rexrayvolumes
```

The `--size=XXX` parameter should specify the storage units (MiB, GiB, etc). It appears to be in GiB.

## Configuration Files
See Above

### Service Log
Nothing of note.

### Client Log

```
root@dcos-agent-private-EEA0354000008:~# rexray volume ls
INFO[0000] updated log level                             logLevel=info
INFO[0000] configured logging                            host=unix:///var/run/libstorage/581843215.sock libstorage.logging.httpRequests=false libstorage.logging.httpResponses=false libstorage.logging.level=info time=1488905187975
INFO[0000] created libStorage client                     clientType=integration disableKeepAlive=false enableInstanceIDHeaders=true enableLocalDevicesHeaders=true host=unix:///var/run/libstorage/581843215.sock logRequests=false logResponses=false lsxPath=/var/lib/libstorage/lsx-linux service= storageDriver=libstorage time=1488905187992
INFO[0000] lsx lock file path                            host=unix:///var/run/libstorage/581843215.sock path=/var/run/libstorage/lsx.lock service= storageDriver=libstorage time=1488905187992
INFO[0000] initializing executors cache                  host=unix:///var/run/libstorage/581843215.sock service= storageDriver=libstorage time=1488905187996
INFO[0000] initializing supported cache                  host=unix:///var/run/libstorage/581843215.sock server=slow-healer-ck service=azure storageDriver=libstorage time=1488905188022
WARN[0000] executor not supported                        host=unix:///var/run/libstorage/581843215.sock server=slow-healer-ck service=azure storageDriver=libstorage time=1488905188089
INFO[0000] successefully dialed libStorage server        host=unix:///var/run/libstorage/581843215.sock server=slow-healer-ck service= storageDriver=libstorage time=1488905188089
INFO[0000] storage driver initialized                    host=unix:///var/run/libstorage/581843215.sock service= storageDriver=libstorage time=1488905188089
INFO[0000] os driver initialized                         host=unix:///var/run/libstorage/581843215.sock osDriver=linux service= storageDriver=libstorage time=1488905188091
INFO[0000] linux integration driver successfully initialized  host=unix:///var/run/libstorage/581843215.sock integrationDriver=linux libstorage.integration.volume.operations.create.default.IOPS= libstorage.integration.volume.operations.create.default.availabilityZone= libstorage.integration.volume.operations.create.default.fsType=ext4 libstorage.integration.volume.operations.create.default.size=16 libstorage.integration.volume.operations.create.default.type= libstorage.integration.volume.operations.create.implicit=true libstorage.integration.volume.operations.mount.path=/var/lib/libstorage/volumes libstorage.integration.volume.operations.mount.rootPath=/data osDriver=linux service= storageDriver=libstorage time=1488905188105
INFO[0000] path cache initializion disabled              host=unix:///var/run/libstorage/581843215.sock integrationDriver=linux osDriver=linux service= storageDriver=libstorage time=1488905188107
INFO[0000] libStorage integration driver successfully initialized  host=unix:///var/run/libstorage/581843215.sock integrationDriver=linux libstorage.integration.volume.operations.create.disable=false libstorage.integration.volume.operations.mount.preempt=false libstorage.integration.volume.operations.path.cache.async=true libstorage.integration.volume.operations.path.cache.enabled=false libstorage.integration.volume.operations.remove.disable=false libstorage.integration.volume.operations.unmount.ignoreusedcount=false osDriver=linux service= storageDriver=libstorage time=1488905188115
INFO[0000] integration driver initialized                host=unix:///var/run/libstorage/581843215.sock integrationDriver=linux osDriver=linux service= storageDriver=libstorage time=1488905188115
INFO[0000] created libStorage client                     host=unix:///var/run/libstorage/581843215.sock integrationDriver=linux osDriver=linux service= storageDriver=libstorage time=1488905188115
WARN[0000] cannot get local deviecs                      error=executor not supported host=unix:///var/run/libstorage/581843215.sock integrationDriver=linux osDriver=linux server=slow-healer-ck service=azure storageDriver=libstorage time=1488905188115 txCR=1488905188 txID=9de61662-5a5b-4eb8-629c-b9754430b527
FATA[0000] http error                                    status=404
```

**Comment by codenrhoden** [comment]
Hi @jzampieron, thanks for the report.

There are a few things to unpack here.

> Step by Step directions with screen shots for configuring an "App Registration" via the Azure Portal

I agree that MS Azure is the hardest to setup by far. Screen shots may be a good addition, but I'd also like to make sure the docs make clear what needs to be accomplished w/o screenshots. The thing about a screen shot is that is out of date as soon as Azure (or anybody else) updates their UI.

> A config.yml for Rexray 0.8.1 or newer that is usable.

The provided YAML in the documentation is indeed intended for libStorage, not REX-Ray. The example YAML is valid for libStorage. A REX-Ray config does not belong in the libStorage documentation. libStorage is intended to have no upstream knowledge of REX-Ray. It looks from the config that you pasted that you figured out how to adapt the config for use within REX-Ray. What could make this easier? Did you happen to find this tool as well: http://rexrayconfig.codedellemc.com/ ? 

> The --size=XXX parameter should specify the storage units (MiB, GiB, etc). It appears to be in GiB.

The [libStorage API](http://docs.libstorage.apiary.io/#reference/volumes/volumes-by-service-collection/create) always uses the size in GiB. We don't allow sizes <1GiB. REX-Ray *could* allow a suffix like 1T to translate to 1024GiB, but I don't think it does that today. I'd have to check. Other than the Apiary documentation, I'm not seeing anything that makes this behavior obvious, so that would be a great addition as well.

Currently, were you able to get the driver working successfully? I think the main thing I am seeing here is difficulty obtaining the required client ID in the form of an app registration? The docs there can certainly be improved. We do accept Pull Requests for enhancements, of course, especially for documentation!

I am also happy to look into introducing `applicationID` as an alias to `clientID`.

**Comment by jzampieron** [comment]
Thanks for the notes. I did manage to get a version of rexray that I built from source with a couple of fixes in place with help from @clintonskitson and @akutz to at least try to work. Then I ran into a brick wall with how our cluster is setup on Azure. 

This is a known, but not well known, limitation of VMSS used in ACS. It's simply _not_ possible to attach a disk to those VMs.

I'm working with some contacts at MSFT to develop an alternate approach.

I think this ticket can be a couple of documentation updates. 

In my efforts, I also discovered that Terraform has some really short and easy directions to making an "App Registration" via the azure cli (the new one). I didn't try the _older_ Azure XPlat CLI.

I didn't try these with rexray, but they did work with terraform.

Link here: https://www.terraform.io/docs/providers/azurerm/index.html
```
az login
az account set --subscription="${SUBSCRIPTION_ID}"
az ad sp create-for-rbac --role="Contributor" --scopes="/subscriptions/${SUBSCRIPTION_ID}"
```
